docs/flamework/additional-modules/components/configuration.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Configuration
+---
+Components have a few features we haven't covered yet!
+
+## Predicate
+You can add a `predicate` function to a component to determine if the component should be constructed for a provided instance.
+
+This only affects Components created via **CollectionService**!
+```ts
+@Component({
+    tag: "Example",
+    predicate: ( instance ) => instance.FindFirstAncestorOfClass("PlayerGui") !== undefined,
+})
+export class ExampleComponent extends BaseComponent<> implements OnStart {}
+```
+
+## Instance Guard
+The `instanceGuard` is similar to `predicate`, but serves a slightly different purpose.
+
+* On the server, an error will be thrown if a component is attempted to be created, but the `instanceGuard` fails.
+* On the client, a connection will be created to wait for `DescendantsAdded` & `DescendantsRemoving` until the `instanceGuard` passes. This is because instances can be sent to the client in pieces and the whole thing might not have arrived yet.
+```ts
+interface Guard extends Tool {
+    Parent: Workspace
+}
+
+const instanceGuard = Flamework.createGuard<Guard>()
+
+@Component( {
+    tag: "Example",
+    instanceGuard: ( instance ): instance is Guard => instanceGuard(instance) && instance.Parent === Workspace,
```

Comment:
Flamework doesn't check any events on ancestry changed and `@rbxts/t` doesn't support checking properties of instances, it'd be better to check parents in `predicate` as the general expectation is that you clone collection service tagged items from storage. This code example is likely from before predicates existed.

Reply:
Hey, thanks for the explanation. I can now see this example makes a lot more sense being turned into a `predicate`.

With that being said, would you mind providing an example you feel would fit here? Unfortunately, this was the only one I could find in Discord and I'm still sorta struggling to understand the correct usage myself.

Reply:
I can't really think of many compelling use cases besides disabling the automatically generated type guards. I believe most things would be better suited as a predicate, at least currently.

You can switch the example to just disable the instanceGuard (`t.any` or `() => true`) and just highlight the fact that it overrides the generated type guards.

Reply:
I believe my most recent commit should address this, let me know if you disagree!